lib/mayaUsd/fileio/jobs/readJob.cpp

```diff
@@ -500,4 +508,13 @@ void UsdMaya_ReadJob::SetMayaRootDagPath(const MDagPath& mayaRootDagPath)
 
 const MDagPath& UsdMaya_ReadJob::GetMayaRootDagPath() const { return mMayaRootDagPath; }
 
+double UsdMaya_ReadJob::timeSampleMultiplier() const { return mTimeSampleMultiplier; }
+
+double UsdMaya_ReadJob::_setTimeSampleMultiplierFrom(const double layerFPS)
+{
+    float sceneFPS = UsdMayaUtil::GetSceneMTimeUnitAsFloat();
```

Comment:
This function is only used in one other place for USD metadata so I can change it to double instead of float.

Is there a list of constant values and/or mathematical formulae to use for correctness? Alternatively I can just update this PR with the changes to double tomorrow and if you can let me know which values should be modified, that would be awesome.

Semi related, it would be awesome if a future API version of MTime provided these out of the box to prevent discrepancies in different implementations that each provide their own mapping.

Reply:
The only non integer framerates that Maya currently supports (24df, 30df, 48df and 60fc) are all based on the same multiplier 1000/1001. So for 24df = 24*1000/1001 ~= 23.976, 30df ~= 29.97 and 60df=60*1000/1001 ~= 59.94
Unless USD does not support the notion of partial frame, rounding can usually and should usually be avoided.

Reply:
@langlor-autodesk Thanks, I updated the PR with a change to `util.cpp/h` to change it all to doubles. 
I also changed all the drop frame conversions to `wholeFrame * 1000.0 / 1001`. 

Hopefully everything looks good now? I was unsure as to the difference between `k29_97FPS` and `k29_97DF` however, so used the same math for both. I hope that's correct.